frontend/javascripts/oxalis/controller/scene_controller.ts

```diff
@@ -430,23 +435,42 @@ class SceneController {
     this.rootNode.add(this.userBoundingBoxGroup);
   }
 
+  private applyTransformToGroup(transform: Transform, group: THREE.Group | CustomLOD) {
+    if (transform?.affineMatrix) {
+      const matrix = new THREE.Matrix4();
+      // @ts-ignore
+      matrix.set(...transform.affineMatrix);
+      // We need to disable matrixAutoUpdate as otherwise the update to the matrix will be lost.
+      group.matrixAutoUpdate = false;
+      group.matrix = matrix;
+    }
+  }
+
   updateUserBoundingBoxesAndMeshesAccordingToTransforms(): void {
     const state = Store.getState();
-    const transformsForGeometries = getTransformsForSkeletonLayer(
+    const tracingStoringUserBBoxes = getSomeTracing(state.annotation);
+    const transformForBBoxes =
+      tracingStoringUserBBoxes.type === "volume"
+        ? getTransformsForLayer(
+            state.dataset,
+            getLayerByName(state.dataset, tracingStoringUserBBoxes.tracingId),
+            state.datasetConfiguration.nativelyRenderedLayerName,
+          )
+        : getTransformsForSkeletonLayer(
+            state.dataset,
+            state.datasetConfiguration.nativelyRenderedLayerName,
+          );
+    this.applyTransformToGroup(transformForBBoxes, this.userBoundingBoxGroup);
+    const visibleSegmentationLayers = getVisibleSegmentationLayers(state);
+    if (visibleSegmentationLayers.length === 0) {
+      return;
+    }
+    const transformForMeshes = getTransformsForLayer(
       state.dataset,
+      visibleSegmentationLayers[0],
       state.datasetConfiguration.nativelyRenderedLayerName,
     );
-    const transformMatrix = transformsForGeometries?.affineMatrix;
-    if (transformMatrix) {
-      const matrix = new THREE.Matrix4();
-      // @ts-ignore
-      matrix.set(...transformMatrix);
-      // We need to disable matrixAutoUpdate as otherwise the update to the matrix will be lost.
-      this.userBoundingBoxGroup.matrixAutoUpdate = false;
-      this.userBoundingBoxGroup.matrix = matrix;
-      this.segmentMeshController.meshesLODRootGroup.matrixAutoUpdate = false;
-      this.segmentMeshController.meshesLODRootGroup.matrix = matrix;
-    }
+    this.applyTransformToGroup(transformForMeshes, this.segmentMeshController.meshesLODRootGroup);
```

Comment:
as we discussed [here](https://github.com/scalableminds/webknossos/pull/8493#issuecomment-2801580769), this will transform meshes of all layers with the transforms of the visible segmentation layer :+1: I think, this is fine if meshes of invisible layers also get hidden automatically (which isn't done in master). does this happen in this pr? I think, we need this for correct mesh rendering.
maybe "refactor SegmentMeshController to grouping meshes by their layer" (quote by you) is necessary for doing the invisible stuff, too? so, one could also do the transform per mesh-layer with not a lot of extra work, but I'm fine with either way :see_no_evil: 
what do you think?

Reply:
Ok sorry, I overlooked the part about making meshes of invisible layers invisible as well. The scene hierarchy should now be adjusted to make this relativly easy. The structure is now: Each layer has its own lod group with the respective meshes and the lod group for each layer can be made invisible on demand